api/notification.js

```diff
@@ -28,22 +28,22 @@ app.get("/", requireAuth, async (req, res) => {
 
 });
 
-// set a notification as checked
-app.patch("/:notificationId", requireAuth, async (req, res) => {
+// delete a notification
+app.delete("/:notificationId", requireAuth, async (req, res) => {
 
   try {
 
     const userId = req.auth.userId;
     const notificationId = validator.toInt(req.params.notificationId);
-    console.log("Check notification", notificationId);
+    console.log("Delete notification", notificationId);
 
-    const results = await checkNotification(notificationId, userId);
+    const results = await deleteNotification(notificationId, userId);
     if (results.affectedRows === 0) {
       console.error("404: No notification found\n");
       res.status(404).send({error: "No notification found."});
     } else {
-      console.log("200: Notification checked\n");
-      res.status(200).send(results);
+      console.log("202: Notification deleted\n");
+      res.status(202).send(results);
```

Comment:
HTTP DELETE should respond with a 204 code if the operation is successful (see [here](https://stackoverflow.com/a/2342589/4048938))

```suggestion
      console.log("204: Notification deleted\n");
      res.status(204).send(results);
```

Reply:
The issue with status code 204 is that it represents 'No Content'. We had agreed earlier that all responses should contain a JSON object and status code 204 prevents the sending of any content.

We had a similar discussion I think when I set the status to 202 for delete plan. However reading that link more, it sounds like 200 would be more appropriate, as 202 represents an operation that is incomplete. If you don't have any objections, I think I will switch to status 200 for all of the delete operations I had set.

Reply:
200 it is!